### PR TITLE
Feature: Column level lineage

### LINKED ISSUE #5325: UI: Add column-level lineage support
We should support column-level lineage in the UI.

### PR DESCRIPTION
**Pull request by ShaileshParmar11**
### Describe your changes :

I worked on adding a column lineage feature 
Closes #5325

#
### Type of change :

- [x] Bug fix
- [x] Improvement
- [x] New feature

#
### Frontend Preview (Screenshots) :

https://user-images.githubusercontent.com/71748675/174569923-fa85bdec-1773-4e01-8b31-1fc533f923de.mov

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

@open-metadata/ui

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomNode.component.tsx [hunk 1]
```diff
@@ -213,23 +229,26 @@ const CustomNode = (props: NodeProps) => {
       ) : null}
       <section
         className={classNames('tw--mx-3 tw-px-3', {
-          'tw-h-36 tw-overflow-y-auto': columns?.length,
+          // 'tw-h-36 tw-overflow-y-auto': !isEmpty(columns),
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomNode.component.tsx:232]: Any reason we have removed this?

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomNode.component.tsx [hunk 2]
```diff
@@ -213,23 +229,26 @@ const CustomNode = (props: NodeProps) => {
...
+                {getHandle(
+                  type,
+                  isConnectable,
+                  isNewNode,
+                  c.fullyQualifiedName
+                )}
+                {getConstraintIcon(c.constraint, 'tw-')}
+                {c.name}
+              </div>
+            </Fragment>
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomNode.component.tsx:250]:
```suggestion
              <div
              key={i}
                className="tw-p-1 tw-rounded tw-border tw-text-grey-body tw-relative"
                data-testid="column">
                {getHandle(
                  type,
                  isConnectable,
                  isNewNode,
                  c.fullyQualifiedName
                )}
                {getConstraintIcon(c.constraint, 'tw-')}
                {c.name}
              </div>
```

Fragment is not required here

File: openmetadata-ui/src/main/resources/ui/src/utils/EntityLineageUtils.tsx [hunk 3]
```diff
@@ -89,9 +82,9 @@ export const getHeaderLabel = (
   );
 };
 
-export const onLoad = (reactFlowInstance: OnLoadParams) => {
+export const onLoad = (reactFlowInstance: ReactFlowInstance) => {
   reactFlowInstance.fitView();
-  reactFlowInstance.zoomTo(1);
+  reactFlowInstance.zoomTo(0.7);
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/utils/EntityLineageUtils.tsx:87]: Having `0.7` as constant will be a good thing

File: openmetadata-ui/src/main/resources/ui/src/utils/EntityLineageUtils.tsx [hunk 4]
```diff
@@ -127,313 +141,181 @@ export const getLineageData = (
...
-        lineageEdges.find((ed: FlowElement) =>
-          (ed as Edge).target.includes(mainNode.id)
-        ) || isEditMode
-          ? lineageEdges.find((ed: FlowElement) =>
-              (ed as Edge).source.includes(mainNode.id)
-            ) || isEditMode
-            ? 'default'
-            : 'output'
-          : 'input',
+      type: getNodeType(entityLineage, mainNode.id),
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/utils/EntityLineageUtils.tsx:250]: Thanks @ShaileshParmar11 for cleanup

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomNode.component.tsx (hunk 1 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomNode.component.tsx:232]: earlier in E2E lineage we were showing columns only clicking on the + icon, but now as we have column lineage we are showing all the column directly so user can connect the C2C lineage

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomNode.component.tsx (hunk 2 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomNode.component.tsx:250]: Thanks for catching 👍

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomNode.component.tsx (hunk 1 above)
Comment (chirag-madlani) [reply openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomNode.component.tsx:232]: Let's remove the line itself if not needed

File: openmetadata-ui/src/main/resources/ui/src/utils/EntityLineageUtils.tsx [hunk 5]
```diff
@@ -114,326 +112,247 @@ export const dragHandle = (event: ReactMouseEvent) => {
...
+  return 'default';
+};
+
+export const getColumnType = (edges: Edge[], id: string) => {
+  const sourceEdge = edges.find((edge) => edge.sourceHandle === id);
+  const targetEdge = edges.find((edge) => edge.targetHandle === id);
+
+  if (sourceEdge?.sourceHandle === id && targetEdge?.targetHandle === id)
+    return 'default';
+  if (sourceEdge?.sourceHandle === id) return 'input';
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/utils/EntityLineageUtils.tsx:142]: Can we define the return values as constants or enum?

**Review by vivekratnavel — APPROVED** [review]

**Pull request merged**